app/src/main/java/org/mozilla/fenix/settings/SyncPreference.kt

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import androidx.appcompat.widget.SwitchCompat
+import androidx.preference.PreferenceViewHolder
+import androidx.preference.SwitchPreferenceCompat
+import org.mozilla.fenix.R
+
+/**
+ * Variation of [SwitchPreferenceCompat] that uses a custom widgetLayoutResource in order to implement
+ * visibility changes to it.
+ * */
+class SyncPreference @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null
+) : SwitchPreferenceCompat(context, attrs) {
+
+    private var switchView: SwitchCompat? = null
+    var widgetVisible: Boolean = false
+
+    init {
+        widgetLayoutResource = R.layout.preference_sync
+    }
+
+    override fun onBindViewHolder(holder: PreferenceViewHolder) {
+        super.onBindViewHolder(holder)
+        switchView = holder.findViewById(R.id.switch_widget) as SwitchCompat
```

Comment:
![image](https://user-images.githubusercontent.com/60002907/116215658-91c8a480-a750-11eb-8e4b-828b766a676e.png)

Reply:
Another question do we need switchView to be nullable?

Reply:
There's smth wrong here, or I'm missing something.
Yes, I believe `switchView` should be nullable as the documentation for `findViewById` says _"@return The view, or {@code null} if no view with the requested ID is found"_. But lint/Android Studio isn't telling me anything about that. I'm also incorrectly casting it as `SwitchCompat` instead of `SwitchCompat?` I've changed that now to cast as `SwitchCompat?`.

🤔